gamebook/game_manager.py

```diff
@@ -1,37 +1,51 @@
 from gamebook.terminal_output import TerminalOutput
 from gamebook.terminal_input import TerminalInput
 import sys
+from gamebook.scene import Scene
 
 
 class GameManager(object):
-    def __init__(self, scene):
-        self.scene = scene
+    def __init__(self, current_scene, scene_import):
+        """
+            takes care of presenting the scenes to the user and moving
+            from one scene to next fluently
+
+            current_scene: the current scene the player is in at the moment
+            when initializing a GameManager, insert the first scene.
+
+            scene_import: a list of all the sceness
+        """
+        self.current_scene = current_scene  # insert the first scene here
+        self.scene_import = scene_import
         self.output_instance = TerminalOutput()
         self.input_instance = TerminalInput()
 
-    def start(self, scene_import):
+    def start(self):
         while True:
             next_scene_name = self.run_scene()
-            self.scene = self.get_next_scene(next_scene_name, scene_import)
+            self.current_scene = self.get_next_scene(next_scene_name)
 
-    def get_next_scene(self, scene_name, scene_import):
-        for scene in scene_import:
+    def get_next_scene(self, scene_name):
+        for scene in self.scene_import:
             if scene.get_name() == scene_name:
                 return scene
 
     def run_scene(self):
+        """evaluates the current scene and returns
+           the following one by it's name"""
 
-        self.output_instance.output(self.scene.show_desc())
+        self.output_instance.output(self.current_scene.show_desc())
 
-        if len(self.scene.options) == 0:
+        if len(self.current_scene.options) == 0:
+            # if the current scene has no options, the game ends
             sys.exit()
```

Comment:
You still need to use the output for this.

Reply:
i did, you commented on an outdated version